src/lib.rs

```diff
@@ -1140,6 +1140,20 @@ impl<'a> Arbitrary<'a> for Ipv6Addr {
     }
 }
 
+impl<'a> Arbitrary<'a> for IpAddr {
+    fn arbitrary(u: &mut Unstructured<'a>) -> Result<Self> {
+        if u.arbitrary()? {
+            u.arbitrary().map(IpAddr::V4)
+        } else {
+            u.arbitrary().map(IpAddr::V6)
+        }
+    }
+
+    fn size_hint(_depth: usize) -> (usize, Option<usize>) {
+        (4, Some(16))
```

Comment:
This version is robust to changes in how many bytes the sub-`Arbitrary` impls consume:

```suggestion
        size_hint::and(
            bool::size_hint(depth),
            size_hint::or(
                Ipv4Addr::size_hint(depth),
                Ipv6Addr::size_hint(depth),
            )
        )
```